aws/aws.go

```diff
@@ -36,98 +36,68 @@ const topicNameMaxLength = 40 - len(topicNamePrefix)
 
-	publisher, pubErr := NewPublisher(cfg, topic)
+	publisher, pubErr := newPublisher(sns.New(sess), topic)
 	if pubErr != nil {
-		return errors.New("error, failed to create a SNS topic: " + pubErr.Error())
+		return pubErr
 	}
-
-	return verifyPermissions(s, publisher, topic, subscriptionID)
+	return verifyPermissions(s, publisher)
 }
 
 // verifyPermissions checks if the aws config has correct permissions
-func verifyPermissions(subscriber pubsub.Subscriber, publisher pubsub.Publisher, topic string, subID string) error {
+func verifyPermissions(subscriber *awsSubscriber, publisher *publisher) error {
 	defer func() {
-		// Delete the publisher and subscriber queue
-		DeletePublisherTopic(topic)
-		DeleteSubscriberByTopic(topic, subID)
+		// Delete publisher topic and subscriber queue
+		publisher.DeleteTopic()
+		subscriber.DeleteSubscription()
 	}()
 
 	ctx, stop := context.WithTimeout(context.Background(), 1*time.Second)
 	defer stop()
-
 	c, e := subscriber.Start(ctx, nil)
 
-	testMessage := []byte("Permissions Verification Test Message.")
+	testMessage := []byte("Permissions Verification Test Message. Id: " + uuid.New().String())
 	errSend := publisher.Publish(ctx, testMessage, nil)
 	if errSend != nil {
-		return errors.New("failed to push message " + errSend.Error())
+		return errSend
 	}
 
 	select {
 	case msg, isOpen := <-c:
 		if !isOpen {
-			return errors.New("error, failed to open message channel")
+			return errors.New("error, channel closed prematurely")
 		}
 		if bytes.Equal(msg.Message(), testMessage) {
 			return nil
 		}
-		return errors.New("error, failed to receive message from publisher")
+		return errors.New("error, received the wrong message from publisher")
 	case err := <-e:
 		return err
 	}
-
-}
-
-// DeletePublisherTopic deletes the SNS topic
-func DeletePublisherTopic(topic string) error {
-	svc := sns.New(session.New())
-
-	paramsDelete := &sns.DeleteTopicInput{
-		TopicArn: aws.String("arn:aws:sns:us-east-1:405093580753:pubsub__" + topic),
-	}
-	_, err := svc.DeleteTopic(paramsDelete)
-	if err != nil {
-		return errors.New("error, failed to delete SNS topic: " + err.Error())
-	}
-	return nil
-}
-
-// DeleteSubscriberByTopic deletes the SQS queue that has given topic and subscription id
-func DeleteSubscriberByTopic(topic string, subID string) error {
-	svc := sqs.New(session.New())
-
-	qURL := "https://sqs.us-east-1.amazonaws.com/405093580753/pubsub__" + topic + "-" + subID
-	paramsDelete := &sqs.DeleteQueueInput{
-		QueueUrl: aws.String(qURL),
-	}
-	_, err := svc.DeleteQueue(paramsDelete)
-	if err != nil {
-		return errors.New("error, failed to delete SQS queue: " + err.Error())
-	}
-	return nil
-
 }
 
 // Utility functions for performing AWS commands (create SNS topic, SQS queue, etc)
 
 // ensureSession will handle creating a session with the passed-in config or a
 // default config if nil was passed in
 func ensureSession(cfg *aws.Config) (*session.Session, error) {
+	_, errConfig := cfg.Credentials.Get()
```

Comment:
hm.. This isn't quite right yet. You can see in the lines below that we're checking for `nil`. This `cfg.Credentials.Get()` call should come after the `nil` check.